lib/server.js

```diff
@@ -136,15 +145,29 @@ module.exports = function createServer(config, log) {
 
             var retryAfter = [blockEmail, blockIpEmail, blockIp].reduce(max)
 
+            var allowIP = ip in allowedIPs.ips
+            var allowedEmailDomain = allowedEmailDomains.isAllowed(email)
+
             if (retryAfter > 0) {
-              if (ip in allowedIPs.ips) {
+              if (allowIP) {
                 retryAfter = 0
               }
-              if (allowedEmailDomains.isAllowed(email)) {
+              if (allowedEmailDomain) {
                 retryAfter = 0
               }
             }
 
+            // IP's that are in blocklist should be blocked
+            // and not return a retryAfter because it is not known
+            // when they would be removed from blocklist
+            if (config.ipBlocklist.enable && blockListManager.contains(ip) && !allowIP && !allowedEmailDomain) {
+              if (!config.ipBlocklist.logOnly) {
+                return {
+                  block: true
+                }
+              }
+            }
```

Comment:
Returning here means we won't call `setRecords`, so we won't write back any updated memcached records for IP addresses on the blocklist.  Your call whether that's important or not, just want to note it explicitly.